includes/admin/extensions/class-extensions-api.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace EDD\Admin\Extensions;
+
+class ExtensionsAPI {
+
+	/**
+	 * Gets the product data from the EDD Products API.
+	 *
+	 * @since 2.11.x
+	 * @param array $body    The body for the API request.
+	 * @param int   $item_id The product ID, if querying a single product.
+	 * @return array|false
+	 */
+	public function get_product_data( $body = array(), $item_id = false ) {
+		if ( empty( $body ) ) {
+			if ( empty( $item_id ) ) {
+				return false;
+			}
+			$body = $this->get_api_body( $item_id );
+		}
+		$key = $this->array_key_first( $body );
+		// The option name is created from the first key/value pair of the API "body".
+		$option_name = "edd_extension_{$key}_{$body[ $key ]}_data";
```

Comment:
```suggestion
		$option_name = sanitize_key( "edd_extension_{$key}_{$body[ $key ]}_data" );
```

Feels like a good idea. Will make sure it's appropriate as an option name.